### PR TITLE
feat: get email template by identifier 

### PR DESCRIPTION
**Pull request by trevorjob**
## Description:

This pull request introduces a new endpoint to retrieve a single email template by its name. The newly added feature allows users to fetch specific email templates from the server, facilitating the use and management of different email templates within the application.

## Changes Made:

- Endpoint Addition: Added a new GET endpoint ``` /api/v1/email/templates/:templateName ``` to fetch a single email template based on - - the provided template name.
- Service Logic: Implemented logic in the ``` EmailService ``` to read and return the requested email template from the file system.
- Error Handling: Added appropriate error handling for scenarios where the requested template does not exist.
- Validation: Ensured that the endpoint correctly handles invalid or missing template names.
- **Endpoint Details:**

- HTTP Method: GET
- Endpoint: /api/v1/email-templates/:name
- URL Parameter:
- name: The name of the email template to be retrieved (e.g.,**welcome**, **reset-password**).
**Request Example:**

GET ``` /api/v1/email-templates/welcome ```
**Response:**

- **Success Response:**

   - Status Code: 200 OK
   - Body:
```python
{
     "status_code": 200,
     "name": "welcome",
     "content": "<html>...</html>"
}
```
   - Description: Returns the requested email template's name and its content.
- **Error Responses:**

   - Template Not Found:

      - Status Code: 404 Not Found
      - Body:
```python
{
  "status_code": 404,
  "message": "Template not found"
}
```
## Motivation and Context:

This feature is essential for fetching specific email templates based on their names, which enhances the flexibility and usability of the email management system. It allows for easier access to individual templates and better integration with other parts of the application that require specific email content.

## How Has This Been Tested?

- Unit Tests: Added tests to ensure that the endpoint correctly retrieves and returns the email template by name.
- Integration Tests: Included tests to verify the interaction between the endpoint and the underlying file system for template retrieval.

## Types of Changes:

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/modules/email/email.controller.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Controller, Get, Param, UseGuards, Request, HttpCode, Query } from '@nestjs/common';
+// import { JwtAuthGuard } from '../auth/jwt-auth.guard';
+import { EmailService } from './email.service';
```
Comment (davidolanrewaju) [inline src/modules/email/email.controller.ts:3]: Please remove all unneccessary comment

File: src/modules/email/email.controller.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+@ApiBearerAuth()
+@ApiTags('Email')
+@Controller('email/templates')
+export class EmailController {
+  constructor(private readonly emailSevice: EmailService) {}
+
+  //   @UseGuards(JwtAuthGuard)
+  @Get(':templateName')
+  @ApiOperation({ summary: 'Get Email Template' })
```
Comment (davidolanrewaju) [inline src/modules/email/email.controller.ts:14]: Remove all unneccessary comment

File: src/modules/email/dto/emailTemplateResponse.ts
Comment (Heba-WebDev) [inline src/modules/email/dto/emailTemplateResponse.ts:1]: Please add decorators to enforce their types

File: src/modules/email/email.controller.ts
Comment (Heba-WebDev) [inline src/modules/email/email.controller.ts:1]: Remove // import { JwtAuthGuard } from '../auth/jwt-auth.guard';

Comment (Heba-WebDev) [reply src/modules/email/email.controller.ts:1]: Remove this as well please,  //   @UseGuards(JwtAuthGuard)

**Comment by Heba-WebDev** [comment]
Left you few comments. You're also failing the CI/CD so please fix that. Good job over all, though.

**Comment by trevorjob** [comment]
@davidolanrewaju @Heba-WebDev 
done.

**Pull request closed without merging**